kswift-gradle-plugin/src/test/kotlin/dev/icerock/moko/kswift/plugin/SealedToSwiftAssociatedEnumFeatureTest.kt

```diff
@@ -0,0 +1,299 @@
+package dev.icerock.moko.kswift.plugin
+
+import dev.icerock.moko.kswift.plugin.context.ClassContext
+import dev.icerock.moko.kswift.plugin.context.LibraryContext
+import dev.icerock.moko.kswift.plugin.feature.Filter
+import dev.icerock.moko.kswift.plugin.feature.SealedToSwiftAssociatedEnumFeature
+import io.outfoxx.swiftpoet.FileSpec
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlinx.metadata.klib.KlibModuleMetadata
+import org.jetbrains.kotlin.library.ToolingSingleFileKlibResolveStrategy
+import org.jetbrains.kotlin.library.resolveSingleFileKlib
+
+@Suppress("LongMethod")
+class SealedToSwiftAssociatedEnumFeatureTest {
+    @Test
+    fun `associated enum feature should produce type mapped output`() {
+        // Generated from TestSealed.kt
+        val klibPath = this::class.java.classLoader.getResource("associated-enum.klib")
+        val konanFile = org.jetbrains.kotlin.konan.file.File(klibPath.toURI().path)
+        // Need to use tooling strategy here since the klib was generated with 1.8
+        // kotlinc-native TestingSealed.kt -p library -o associated-enum
+        val library = resolveSingleFileKlib(
+            libraryFile = konanFile,
+            strategy = ToolingSingleFileKlibResolveStrategy,
+        )
+        val metadata = KlibModuleMetadata.read(KotlinMetadataLibraryProvider(library))
+        val libraryContext = LibraryContext(metadata)
+        val fileSpecBuilder = FileSpec.builder(moduleName = "module", fileName = "file")
+
+        libraryContext.visit { featureContext ->
+            (featureContext as? ClassContext)?.let {
+                SealedToSwiftAssociatedEnumFeature(
+                    featureContext = ClassContext::class,
+                    filter = Filter.Exclude(emptySet()),
+                ).doProcess(
+                    featureContext = it,
+                    fileSpecBuilder = fileSpecBuilder,
+                    kotlinFrameworkName = "shared",
+                )
+            }
+        }
+
+        val appendable = InMemoryAppendable()
+        fileSpecBuilder.build().writeTo(appendable)
+        val expected = """import Foundation
+import shared
+
+/**
+ * selector: ClassContext/associatedenum/com/icerockdev/library/associatedenum/TestingSealed */
+public enum TestingSealedKs {
```

Comment:
so, as i see, this feature not about swift's associated types usage but improved version of SealedToEnum with correct pass of sealed internal data to enum internal data, yes? looks good!